src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -41,45 +42,143 @@ typedef unsigned long long int  fmi3_uint_buf_t; /* same size as fmi3_int_buf_t
 #define UINTXX_MIN (0)
 #define FMI3_FLOAT32_MAX (3.402823466e+38F) /* FLT_MAX from float.h is platform-dependent (doesn't have to be 32 bit) */
 
+// TODO: Move these + access functions into the parser_scheme file?
 #define ATTR_STR(attr) #attr,
-const char *fmi3_xmlAttrNames[fmi3_xml_attr_number] = {
-    FMI3_XML_ATTRLIST(ATTR_STR)
+const jm_string fmi3_modelDescription_xmlAttrNames[fmi3_modelDescription_xml_attr_number] = {
+    FMI3_XML_ATTRLIST_MODEL_DESCR(ATTR_STR)
 };
 
+const jm_string fmi3_termIcon_xmlAttrNames[fmi3_xml_termIcon_attr_number] = {
+    FMI_XML_ATTRLIST_TERM_ICON(ATTR_STR)
+};
+
+static jm_string fmi3_xml_get_xml_attr_name(fmi3_xml_parser_context_t* context, const fmi3_xml_attr_t enu) {
+    const fmi3_xml_type_t xmlType = context->xmlType;
+    switch (xmlType) {
+        case fmi3_xml_type_modelDescription:
+            return fmi3_modelDescription_xmlAttrNames[enu.modelDescription];
+        case fmi3_xml_type_terminalAndIcons:
+            return fmi3_termIcon_xmlAttrNames[enu.termIcon];
+    }
+    return "unknown";
+}
+
 /**
  * The expansion of below macro is also a macro. Example:
  * EXPAND_ELM_SCHEME(Float64) -> fmi3_xml_scheme_Float64 -> {fmi3_xml_elmID_SimpleType, fmi3_xml_elmID_TypeDefinitions, 0, 1}
  */
 #define EXPAND_ELM_SCHEME(elm) fmi3_xml_scheme_##elm ,
-
-/* Global array of all scheme_info_t. Index it with fmi3_xml_elm_enu_t entries. */
-fmi3_xml_scheme_info_t fmi3_xml_scheme_info[fmi3_xml_elm_number] = {
+/* Global array of all modelDescription scheme_info_t. Index it with fmi3_xml_elm_modelDescription_enu_t entries. */
+const fmi3_xml_scheme_modelDescription_info_t fmi3_xml_scheme_modelDescription_info[fmi3_xml_elm_number] = {
     FMI3_XML_ELMLIST_MODEL_DESCR(EXPAND_ELM_SCHEME)
-    FMI_XML_ELMLIST_TERM_ICON   (EXPAND_ELM_SCHEME)
-    {fmi3_xml_elm_actual_number,0,0},
+    {fmi3_xml_elm_actual_number, 0, 0},
     FMI3_XML_ELMLIST_ALT_MODEL_DESCR     (EXPAND_ELM_SCHEME)
-    FMI_XML_ELMLIST_ALT_TERM_ICON        (EXPAND_ELM_SCHEME)
     FMI3_XML_ELMLIST_ABSTRACT_MODEL_DESCR(EXPAND_ELM_SCHEME)
 };
 
+#define EXPAND_ELM_SCHEME_TERMICON(elm) fmi_xml_scheme_termIcon_##elm ,
+/* Global array of all terminalsAndIcons scheme_info_t. Index it with fmi_xml_elm_termIcon_enu_t entries. */
+fmi_xml_scheme_termIcon_info_t fmi_xml_scheme_termIcon_info[fmi_xml_elm_termIcon_number] = {
+    FMI_XML_ELMLIST_TERM_ICON(EXPAND_ELM_SCHEME_TERMICON)
+    {fmi_xml_elm_termIcon_actual_number, 0, 0},
+    FMI_XML_ELMLIST_ALT_TERM_ICON(EXPAND_ELM_SCHEME_TERMICON)
+};
+
+// TODO: Move to more suitable place?
+static fmi3_xml_scheme_info_t fmi3_xml_get_scheme_info(fmi3_xml_parser_context_t* context, fmi3_xml_elm_t enu) {
+    const fmi3_xml_type_t xmlType = context->xmlType;
+
```

Comment:
Contents of this function likely a bit more "verbose" than strictly required, but I don't necessarily think this is an issue; makes the conversion very explicit.